src/components/datepicker/js/datepickerDirective.js

```diff
@@ -402,9 +402,17 @@
 
     // Responds to external changes to the model value.
     self.ngModelCtrl.$formatters.push(function(value) {
+      var parsedValue = angular.isDefined(value) ? Date.parse(value) : null;
+
+      if (parsedValue) {
```

Comment:
Whoops, I inverted the equality in the expression. Note that `angular.isNumber(NaN)` returns `true` so it should be checked second. For example `if (!isNaN(parsedValue) && angular.isNumber(parsedValue))`